### PR TITLE
Implement Tests data sources

### PR DESCRIPTION
**Pull request by Fenthick**
Issue: KNTK-143

Based on: #4

### CODE REVIEW

**Review by mateuszmidor — APPROVED** [review]

**Review by lukaa12 — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]
OK, to go in, but I'd like the examples/demos to be combined in the future.

File: synthetics/data_source_test_impl.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/kentik/community_sdk_golang/apiv6/kentikapi"
+	"github.com/kentik/community_sdk_golang/apiv6/kentikapi/synthetics"
+)
+
+func dataSourceTest() *schema.Resource {
+	return &schema.Resource{
+		Description: "DataSource representing single synthetic test test",
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply synthetics/data_source_test_impl.go:17]: I'll fix it in next PR.

File: synthetics/data_source_test_test.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package synthetics
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// Note: values checked in tests below are provided by stub API Server from test-data.json (running in background)
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply synthetics/data_source_test_test.go:9]: I'll try to improve it in one of next PRs.

File: examples/data-sources/kentik-synthetics_tests/run.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply examples/data-sources/kentik-synthetics_tests/run.sh:1]: Are you talking about removing those 2 examples? I think that showing how to use those data sources is still helpful - it show that this functionality is available.
GET requests performed by `kentik-synthetics_test` TF resource is a different part of code than `kentik-synthetics_test[s]` TF data sources.

File: synthetics/data_source_agents_test.go [hunk 4]
```diff
@@ -13,59 +13,57 @@ func TestDataSourceAgents(t *testing.T) {
 	resource.UnitTest(t, resource.TestCase{
 		PreCheck:          func() { checkAPIServerConnection(t) },
 		ProviderFactories: providerFactories(),
-		Steps: []resource.TestStep{
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply synthetics/data_source_agents_test.go:16]: Those are tests for 2 different data sources.  TF providers style requires to have one file per data source. So we have `data_source_agent.go` and `data_source_agents.go` files. I kept that separation for test files as well. I followed the structure of cloudexport provider for such things.

**Pull request merged**